docker/bitcoind/start-bitcoind.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# exit from script if error was raised.
+set -e
+
+# error function is used within a bash function in order to send the error
+# message directly to the stderr output and exit.
+error() {
+    echo "$1" > /dev/stderr
+    exit 0
+}
+
+# return is used within bash function in order to return the value.
+return() {
+    echo "$1"
+}
+
+# set_default function gives the ability to move the setting of default
+# env variable from docker file to the script thereby giving the ability to the
+# user override it during container start.
+set_default() {
+    # docker initialized env variables with blank string and we can't just
+    # use -z flag as usually.
+    BLANK_STRING='""'
+
+    VARIABLE="$1"
+    DEFAULT="$2"
+
+    if [[ -z "$VARIABLE" || "$VARIABLE" == "$BLANK_STRING" ]]; then
+
+        if [ -z "$DEFAULT" ]; then
+            error "You should specify default variable"
+        else
+            VARIABLE="$DEFAULT"
+        fi
+    fi
+
+   return "$VARIABLE"
+}
+
+# Set default variables if needed.
+RPCUSER=$(set_default "$RPCUSER" "devuser")
+RPCPASS=$(set_default "$RPCPASS" "devpass")
+NETWORK=$(set_default "$NETWORK" "regtest")
+DEBUG=$(set_default "$BITCOIND_DEBUG" "1")
+
+PARAMS=""
+if [ "$NETWORK" != "mainnet" ]; then
+   PARAMS="-$NETWORK"
+fi
+
+PARAMS=$(echo $PARAMS \
+    "-debug"="$DEBUG" \
+    "-rpcuser"="$RPCUSER" \
```

Comment:
Any reason this is using rpcuser/rpcpassword rather than rpcauth or cookie based authentication?

Reply:
To use rpcauth you need to first hash the password for which you need to have the appropriate tooling available in the shell script. And cookie based auth would require files to be shared between docker containers while maintaining the proper permissions, which in my experience isn't worth the hassle.
Maybe we should make it more clear that these scripts/containers/composes are just meant as a quick development setup and shouldn't be used in production.

Reply:
Sure, my point is more than at some point user/pass will be removed from Core (they are mostlyconsidered deprecated), and then the alternative methods will need to be used in either case, so it seems beenficial to start using them now.

Reply:
Hmm, yeah, I keep forgetting that it's deprecated. I guess we can just replace `$RPCUSER` and `$RPCPASSWORD` with `$RPCAUTH` and leave the hashing of the password to the user that wants to change the values (and perhaps link to the [python script](https://github.com/bitcoin/bitcoin/blob/master/share/rpcauth/rpcauth.py) that can generate it.

Reply:
Okay ACK will update the PR accordingly! That said I've noticed that `lnd` doesn't parameter for `bitcoind.rpcauth` ?
```
bitcoind:
      --bitcoind.dir=                                        The base directory that contains the
                                                             node's data, logs, configuration
                                                             file, etc. (default:
                                                             /Users/mohamedawnallah/Library/Applic-

                                                             ation Support/Bitcoin)
      --bitcoind.config=                                     Configuration filepath. If not set,
                                                             will default to the default filename
                                                             under 'dir'.
      --bitcoind.rpccookie=                                  Authentication cookie file for RPC
                                                             connections. If not set, will default
                                                             to .cookie under 'dir'.
      --bitcoind.rpchost=                                    The daemon's rpc listening address.
                                                             If a port is omitted, then the
                                                             default port for the selected chain
                                                             parameters will be used. (default:
                                                             localhost)
      --bitcoind.rpcuser=                                    Username for RPC connections
      --bitcoind.rpcpass=                                    Password for RPC connections
      --bitcoind.zmqpubrawblock=                             The address listening for ZMQ
                                                             connections to deliver raw block
                                                             notifications
      --bitcoind.zmqpubrawtx=                                The address listening for ZMQ
                                                             connections to deliver raw
                                                             transaction notifications
      --bitcoind.zmqreaddeadline=                            The read deadline for reading ZMQ
                                                             messages from both the block and tx
                                                             subscriptions (default: 5s)
      --bitcoind.estimatemode=                               The fee estimate mode. Must be either
                                                             ECONOMICAL or CONSERVATIVE. (default:
                                                             CONSERVATIVE)
      --bitcoind.pruned-node-max-peers=                      The maximum number of peers lnd will
                                                             choose from the backend node to
                                                             retrieve pruned blocks from. This
                                                             only applies to pruned nodes.
                                                             (default: 4)
      --bitcoind.rpcpolling                                  Poll the bitcoind RPC interface for
                                                             block and transaction notifications
                                                             instead of using the ZMQ interface
      --bitcoind.blockpollinginterval=                       The interval that will be used to
                                                             poll bitcoind for new blocks. Only
                                                             used if rpcpolling is true.
      --bitcoind.txpollinginterval=                          The interval that will be used to
                                                             poll bitcoind for new tx. Only used
                                                             if rpcpolling is true.
```

Reply:
That's correct. `lnd` still needs to know the actual password since it needs to send it over the RPC. The `rpcauth` config option in `bitcoind` just allows users to not need to specify a clear text password in `bitcoin.conf`. But any RPC client still needs to know the clear text password (or use the cookie authentication instead).